### ISSUE TITLE
Dynamic import doesn't work with generators

### ISSUE THREAD

**Issue by nowindowsowrking** [jquery/esprima#1781]
Implemented dynamic import support in Esprima does not work with generators. 

### Steps to reproduce

```js
esprima.parse((`function* a() {
  yield import('http');
}`)
```

### Expected output

Should parse

### Actual output

Throws an syntax error

### Relevant references
A relevant reference has to be the similiar test case for [Babylon](https://github.com/babel/babylon/tree/master/test/fixtures/experimental/dynamic-import/generator)

**Comment by michaelficarra** [comment]
Please don't refer to future ECMAScript features as ES20XX unless they have reached stage 4 and been incorporated into the appropriate draft spec. Until then, they may never make it into the specification, let alone a particular year.

**Comment by nowindowsowrking** [comment]
@michaelficarra I am aware of this, but this was reported because this feature have been implemented into Esprima as seen [here](https://github.com/jquery/esprima/pull/1740)

**Comment by mikesherov** [comment]
Thanks for contributing, @nowindowsowrking! As usual, you catch everything. @michaelficarra's was simply that you should refer to them as "2018 dynamic imports", not that the report was invalid.

**Comment by nowindowsowrking** [comment]
@mikesherov Removed 2018 reference. BTW. There are also a few other issues with other features implemented belonging to the upcoming 5.0 version. I am right if I say there are no point to report them because the specs will probably change?

**Comment by mikesherov** [comment]
You can and should definitely report issues for code that is in the Esprima master branch.
